routers/api/v1/utils/hook.go

```diff
@@ -164,13 +165,20 @@ func addHook(ctx *context.APIContext, form *api.CreateHookOption, ownerID, repoI
 	if len(form.Events) == 0 {
 		form.Events = []string{"push"}
 	}
+
+	isSystemWebhook, err := strconv.ParseBool(form.Config["is_system_webhook"])
+	if err != nil {
+		ctx.Error(http.StatusUnprocessableEntity, "", "Invalid is_system_webhook value")
```

Comment:
What is `is_system_webhook`? I didn't see any document for it.

And I do not think `err != nil` is right here, because `strconv.ParseBool("")` returns an error: is `is_system_webhook` optional?

The last question, what is `form.Config`? Is it from a JSON? Is there a chance to use "bool" type?

Reply:
> What is is_system_webhook? I didn't see any document for it.

I haven't add the document about `is_system_webhook`, but I noticed that when we create system hooks in the gitea admin page, it will write the `is_system_webhook` column in `webhook` table.
![image](https://github.com/go-gitea/gitea/assets/38434877/02797be1-41f4-4175-a8ff-898870249e39)
And when we create webhooks via `/api/v1/admin/hooks` API, we can not choose what type of webhook to create. We can only create `default_webhook`. So I want add this field to this API.

>And I do not think err != nil is right here, because strconv.ParseBool("") returns an error: is is_system_webhook optional?

Thanks, You are right. I have optimize the code .

>The last question, what is form.Config? Is it from a JSON? Is there a chance to use "bool" type?

Yes, it's a `map[string]string`, if we change it to `map[string]any`, a lot of codes should be change.